fitbenchmarking/controllers/tests/test_controllers.py

```diff
@@ -237,6 +237,22 @@ def test_eval_conf(self):
 
         self.assertAlmostEqual(controller.eval_confidence(), 0.192, 6)
 
+    @patch("fitbenchmarking.controllers.base_controller.curve_fit")
+    def test_eval_conf_failed_fit(self, mock):
+        """
+        BaseSoftwareController: Test eval_confidence function handles
+        RuntimeError correctly
+        """
+        controller = DummyController(self.cost_func)
+        controller.params_pdfs = {'A1': [4, 4, 4, 4, 4],
+                                  'A2': [3, 3.7, 3, 3, 3],
+                                  'A3': [2, 2, 2, 2.4, 2.5],
+                                  'A4': [0.5, 0.7, 1, 1, 1.2]}
+        mock.side_effect = RuntimeError
+        acc = controller.eval_confidence()
+        self.assertEqual(acc, 0)
+        assert controller.flag == 8
```

Comment:
is there a reason you didnt use `self.asssertEquals(controller.flag, 8)`?

Reply:
I think I just copied the flag check from other tests within `test_controllers`, but agree it would be better to use `assertEquals` so will update this